### PR TITLE
Add opam-lib 1.3.0

### PR DESCRIPTION
**Pull request by AltGr**
Hopefully this can be a first, easier step to migrate the tooling to
2.0; this version is still fully compatible with the 1.2.2 format
repository, but already has a few of the improvements of opam 2, and
e.g. a more complete opam lint.

This is intended to be the new base for the tooling (e.g. Camelus,
opam2web...) while the repository is still on opam 1.2.2; it will
also, later, be needed for the 2.0.0 to 1.2.2 format downgrade tool)

### CODE REVIEW

**Comment by avsm** [comment]
awesome! could we get a quick blog post about this one on platform-blog perhaps? that would alert all the plugin authors

**Comment by avsm** [comment]
Also do we need upper bounds on existing opam-lib consumers if this changes the API?

**Comment by AltGr** [comment]
Yes, it breaks parts of the API; I have a pending PR for opamfu, am doing opam2web.
Camelus is already on 1.3.

**Comment by AltGr** [comment]
(the 2.0 dev lib is already available though, so I may as well document that in the same post to clarify)

**Comment by AltGr** [comment]
Done: https://github.com/ocaml/platform-blog/pull/39

**Comment by avsm** [comment]
> Yes, it breaks parts of the API; I have a pending PR for opamfu, am doing opam2web. Camelus is already on 1.3.

Great! I meant for existing released packages -- an upper bound as part of this pr on opamfu/opam2web etc.

**Comment by AltGr** [comment]
I don't get exactly what travis does, but the 1.2.0 test failed with topkg-care on opam-lib 1.2.2, after having successfully installed 1.3.0 then removed it... it doesn't seem related to the PR. As for the MacOS build, it hit a timeout.

File: packages/topkg-care/topkg-care.0.8.0/opam [hunk 1]
```diff
@@ -18,10 +18,10 @@ depends: [
   "bos"
   "cmdliner"
   "webbrowser"
-  "opam-lib" {>= "1.2.2"}
+  "opam-lib" {= "1.2.2"}
 ]
 build:[
   "ocaml" "pkg/pkg.ml" "build"
           "--pkg-name" name
```
Comment (samoht) [inline packages/topkg-care/topkg-care.0.8.0/opam]: this needs quoting for opam 1.2.0 to be happy: `%{name}%` /cc @dbuenzli

**Comment by samoht** [comment]
Thanks!

**Pull request merged**